pkg/services/accesscontrol/scope.go

```diff
@@ -123,18 +123,18 @@ func (s *ScopeResolver) GetResolveKeywordScopeMutator(user *models.SignedInUser)
 	}
 }
 
-type AttributeScopeResolveFunc func(ctx context.Context, orgID int64, initialScope string) (string, error)
+type AttributeScopeResolveFunc func(ctx context.Context, user *models.SignedInUser, initialScope string) (string, error)
 
 // getCacheKey creates an identifier to fetch and store resolution of scopes in the cache
 func getCacheKey(orgID int64, scope string) string {
 	return fmt.Sprintf("%s-%v", scope, orgID)
 }
 
 // GetResolveAttributeScopeMutator returns a function to resolve scopes with attributes such as `name` or `uid` into `id` based scopes
-func (s *ScopeResolver) GetResolveAttributeScopeMutator(orgID int64) ScopeMutator {
+func (s *ScopeResolver) GetResolveAttributeScopeMutator(user *models.SignedInUser) ScopeMutator {
```

Comment:
Feels a bit weird to need the user but have a cache key independant of it. I wonder if it could lead to security issues?

Reply:
You're right, I hadn't really thought of that properly. Wouldn't have been a problem with this change, but might have caused issues in the future. I've changed the approach to use a mock user from the annotation type resolver. Let me know what you think!